### PR TITLE
Feature | API Playground (WIP)

### PR DESCRIPTION
**Pull request by kyrylolvov**
What has been changed:

[4c4439e713c6a35498147d89dca9edb06d4d80f6](https://github.com/fuma-nama/fumadocs/pull/565/commits/4c4439e713c6a35498147d89dca9edb06d4d80f6)
Modified how APIInfo component looks and added ability to copy endpoint url.
Added InteractiveButton component to API components
Added ApiContext and added it to RootProvider

[593991f0a645885aae779af176f80dcbe9a26db8](https://github.com/fuma-nama/fumadocs/pull/565/commits/593991f0a645885aae779af176f80dcbe9a26db8)
Updated APIInfo component, so it can be used without passing children
Added SendButton component and its variants

[1ae42404d654d7c666d07aa95a06e79e7193df48](https://github.com/fuma-nama/fumadocs/pull/565/commits/1ae42404d654d7c666d07aa95a06e79e7193df48)
Added statusMap, which is used to get most used response status descriptions
Added getStatusInfo, which returns general information about response types depending on the range

[1df0e2c2609f846c99adbddae64d232c067164b2](https://github.com/fuma-nama/fumadocs/pull/565/commits/1df0e2c2609f846c99adbddae64d232c067164b2)
Added CodeBlock component, which renders API response during runtime

[c64e5ae794c142783150481622bbfa09f51c42ad](https://github.com/fuma-nama/fumadocs/pull/565/commits/c64e5ae794c142783150481622bbfa09f51c42ad)
Added shadcn input, label, form components and related radix-ui dependencies
Added APIPlayground and ArrayInput components

[bce3f1050f90a7ec743518a4581d8187c5c2d9e3](https://github.com/fuma-nama/fumadocs/pull/565/commits/bce3f1050f90a7ec743518a4581d8187c5c2d9e3)
Updated RootProvider for docs to use correct API url
Fixed TypeScript errors for APIPlayground
Added basic documentation

### CODE REVIEW

**Comment by kyrylolvov** [comment]
https://github.com/fuma-nama/fumadocs/assets/70720713/d6684561-fe93-4663-b4f9-f3d9b18fef05

**Comment by kyrylolvov** [comment]
@fuma-nama Thats basically the addition of ApiPlayground, just need to have types fixed in there, but its pretty much ready for review. I have developed it and tested in the docs I do, and everything works.

**Comment by fuma-nama** [comment]
Thanks for contributing the API playground! I'm excited to hear that

**Comment by kyrylolvov** [comment]
@fuma-nama I have fixed the TypeScript errors, but its better if you take a look at that. Added docs for Playground and marked it as experimental, as some features which users might need, are not implemented (like custom fields).

Also, I have noticed a bug, which I dont really know how to fix. The isLoading and isValidating from useSwr doesnt change. If you know and have time, could you take a look, as I'm not that familiar with SWR, I mostly use Tanstack-Query.

**Comment by fuma-nama** [comment]
Also some build problems here, maybe you can run the production build on your own to reproduce the problem
image

**Comment by fuma-nama** [comment]
I'll work on the API playground

**Comment by kyrylolvov** [comment]
@fuma-nama Just found out how to fix the loading state. Need to update useSwr hook.

```
const {
    data,
    isLoading,
    isValidating,
    mutate
} = useSWR(
    `${method}-${baseUrl}-${route}`,
    () => fetcher(form.getValues()), {
        revalidateOnMount: false,
        shouldRetryOnError: false,
        revalidateOnFocus: false,
    },
);
```

**Comment by fuma-nama** [comment]
I see. You don't have to make changes now, I'm rewriting some code

File: packages/openapi/test/out/museum/operations.mdx
Comment (kyrylolvov) [inline packages/openapi/test/out/museum/operations.mdx:1]: For baseUrl, better to set `https://redocly.com/_mock/docs/openapi/museum-api`. For some reason the fake-museum-example.com doesn't work

Comment (kyrylolvov) [reply packages/openapi/test/out/museum/operations.mdx:1]: https://github.com/fuma-nama/fumadocs/assets/70720713/f4651724-b7cb-4c5c-96c2-8c79a9d3fe08

File: packages/openapi/test/out/museum/events.mdx
Comment (kyrylolvov) [inline packages/openapi/test/out/museum/events.mdx:1]: Might be a better thing to have authorization set to "Basic Og==" or just "Basic", as Redocly's fake API uses Basic, not Bearer authorization

**Pull request merged**